prov/gni/include/gnix_nic.h

```diff
@@ -259,6 +285,7 @@ int _gnix_nic_tx_free(struct gnix_nic *nic, struct gnix_tx_descriptor *tdesc);
  * @brief allocate a gnix_nic struct
  *
  * @param[in] domain   pointer to previously allocated gnix_fid_domain struct
+ * @param[in] type     specify whether shared or dedicate nic be allocated
```

Comment:
More importantly, these comments don't match the function signature that follows.